elements/map/src/interaction.ts

```diff
@@ -0,0 +1,36 @@
+import * as interactions from "ol/interaction";
+import Map from "ol/Map.js";
+
+export type InteractionsType =
+  | [
+      "DragRotate",
+      "DoubleClickZoom",
+      "DragPan",
+      "PinchRotate",
+      "PinchZoom",
+      "KeyboardPan",
+      "KeyboardZoom",
+      "MouseWheelZoom",
+      "DragZoom"
+    ]
+  | [];
+
+/**
+ * Loop through interactions available on map
+ * and setActive to `false
+ * `
+ * @param {InteractionsType} disableInteractions
+ * @param {Map} olMap
+ */
+export function setInteractionInactive(
+  disableInteractions: InteractionsType,
+  olMap: Map
+) {
+  olMap.getInteractions().forEach(function (interaction) {
+    disableInteractions.forEach(
+      (type) =>
+        interaction instanceof interactions[type] &&
+        interaction.setActive(false)
+    );
+  });
```

Comment:
> If this is only used for disabling interactions while scrolling, you can probably catch the corresponding event and then disable all interactions with `eoxMap.map.getInteractions().forEach(i => i.setActive(false))`.

Currently, I am using the same logic of `setInteraction()` by using `olMap` i.e. `EOx.map`.